### PR TITLE
[ONNX] Fix ShuffleNetV2 model export issue.

### PR DESCRIPTION
**Pull request by fatcat-z**
As described in Issue [#48093](https://github.com/pytorch/pytorch/issues/48093), torch.onnx.export() method failed to export a correct ONNX file for ShuffleNetV2 model while dynamic_axes parameter was provided.

This PR fixes this issue by changing the way to read parameters in channel_shuffle() method.

### CODE REVIEW

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

Can you also add some basic ONNX tests for shufflenet in https://github.com/pytorch/vision/blob/master/test/test_onnx.py so that we can track this?

**Comment by fatcat-z** [comment]
@fmassa Thanks for your comments. The test has been added.

File: torchvision/models/shufflenetv2.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 
 def channel_shuffle(x: Tensor, groups: int) -> Tensor:
-    batchsize, num_channels, height, width = x.data.size()
+    batchsize, num_channels, height, width = x.shape[0], x.shape[1], x.shape[2], x.shape[3]
```
Comment (BowenBao) [inline torchvision/models/shufflenetv2.py:22]: Thanks for spotting, this is the root cause of sizes exported as constant, correct?
Can you give a quick try if `x.size()` works? If so we can keep the change minimal. We should definitely add a section in onnx doc page about these known issues of things getting exported as constant :)

Comment (fatcat-z) [reply torchvision/models/shufflenetv2.py]: Yes, x.size() also works and I've updated the code accordingly.

**Review by datumbox — APPROVED** [review]
@fatcat-z LGTM! Thank you!

I tested your unit-test without the patch and it [fails as expected](https://app.circleci.com/pipelines/github/pytorch/vision/5489/workflows/1398d94b-2b3a-4798-8f81-2c65aba6b051/jobs/346025). Thanks for submitting the fix.

File: test/test_onnx.py [hunk 2]
```diff
@@ -482,6 +482,17 @@ def test_keypoint_rcnn(self):
                        dynamic_axes={"images_tensors": [0, 1, 2]},
                        tolerate_small_mismatch=True)
 
+    def test_shufflenet_v2_dynamic_axes(self):
+        model = models.shufflenet_v2_x0_5(pretrained=True)
+        dummy_input = torch.randn(1, 3, 224, 224, requires_grad=True)
+        test_inputs = torch.cat([dummy_input, dummy_input, dummy_input], 0)
```
Comment (jiafatom) [inline test/test_onnx.py:488]: Is there a particular reason to concatenate `dummy_input` to `test_inputs`? Can I use `test_inputs = torch.randn(3, 3, 224, 224, requires_grad=True)`?

Comment (fmassa) [reply test/test_onnx.py:488]: @fatcat-z any thoughts on this? Is it so that we can compare the output by passing `dummy_input` and `test_inputs` (as they are equal on both dimensions)?
